### ISSUE TITLE
Unable to load video with short byterange

### ISSUE THREAD

**Issue by stevenviola** [videojs/videojs-contrib-hls#1279]
## Description

I'm serving up various types of video and occasionally, the first segment needs to have a byterange set so that it starts at a particular time. On a particular small file, the amount of the first segment's byterange that needs to be served comes out to 12kB, but that results in an error: _The media could not be loaded, either because the server or network failed or because the format is not supported._

Expanding the size of the byterange so that the size is 32kB on the same file works, and plays back with no issue. The format and codec stay the same, so I the file is not corrupted, just byterange and duration have changed. Here's a jsbin for the shorter not working example, and the working example with the larger byterange

**Not Working:** http://jsbin.com/tedufoguno/edit?html,output
**Working:** http://jsbin.com/tedufoguno/edit?html,output

Using curl to request the same range for the _Not Working_ content results in a ts file which can be played back:

```
curl 'https://s3-us-west-2.amazonaws.com/cm-public-web/byteserve/20171012155750.ts' -H 'Range: bytes=201724-213755' -o file.ts
```

I'm able to re-encode this range with ffmpeg and not have any errors.

## Sources
Bytes `201724-213755` of [this file](https://s3-us-west-2.amazonaws.com/cm-public-web/byteserve/20171012155750.ts) are unable to be played back if they're first in the manifest. Bytes `179540-213755` do play back as expected. In each of those ranges, the first video packet is the keyframe. The video is a still black frame which is why the video file size is so low.

## Steps to reproduce
Explain in detail the exact steps necessary to reproduce the issue.

1. Load https://s3-us-west-2.amazonaws.com/cm-public-web/byteserve/not-working.m3u8 in https://videojs.github.io/videojs-contrib-hls/
2. Compare that m3u8 with https://s3-us-west-2.amazonaws.com/cm-public-web/byteserve/working.m3u8 which works

### Expected

The range of the video file is loaded and played back without issue

### Error output
```VIDEOJS: ERROR: (CODE:4 MEDIA_ERR_SRC_NOT_SUPPORTED) The media could not be loaded, either because the server or network failed or because the format is not supported. MediaError {code: 4, message: "The media could not be loaded, either because the …rk failed or because the format is not supported."}```

## Additional Information:

Comparing against a [larger video](https://s3-us-west-2.amazonaws.com/cm-public-web/byteserve/larger-file.m3u8), for a similar duration, the larger video file plays back with no issue. In both cases, inspecting the video shows that they each only have one video key frame as the first packet, yet the smaller range doesn't play back while the larger video does play back.

### videojs-contrib-hls version
videojs-contrib-hls 5.11.1

### videojs version
video.js 6.2.8

### Browsers
`what browsers are affected? please include browser and version for each`
Behavior observed in the following browsers on Mac OS X

* Chrome 61
* Safari 10.1.2

**Comment by mjneil** [comment]
Looks like since the PAT and PMT packets are not the first packets of the segment, mux.js attempts to cache the initial video packets until it sees the PMT, and then processes those video packets. However it looks like mux.js is processing those packets too soon, before it can setup the rest of the pipeline with the data from the PMT.

**Comment by stevenviola** [comment]
@mjneil, thanks for looking into this so quickly. Is there a better range I should specify from the source file that's not working? I'm using ffmpeg to iterate through the packets of the video file to find the video keyframe that is nearest to the desirable time. This has been working well so far and this was the first time I've had an issue with videojs not loading a video

**Comment by mjneil** [comment]
No what you are doing currently is fine. The data for the segment you are sending has everything needed, PAT and PMT packets and a complete GOP. Your data is a bit outside of the HLS spec recommendation though. `The first two Transport Stream packets in a Segment without an EXT-X-MAP tag SHOULD be a PAT and a PMT.` In your case, the PAT and PMT packets are in the middle. While it goes against the recommendation, it is still valid. You could always ensure the PAT and PMT are the first two packets, which should correct the problem. However, since your segment is still valid, this is a bug in mux.js not properly processing the packets that come before the PAT and PMT. I'll update this issue when a fix has made it into a contrib-hls release. Maybe sometime next week, though I don't have an accurate ETA at the moment.

**Comment by mjneil** [comment]
The fix for this has been released in v5.12.0. Closing this issue for now, please re-open or submit a new one if you continue to experience issues.

**Comment by stevenviola** [comment]
@mjneil Thanks a lot. This looks like it fixed my issue!
